src/hotspot/cpu/riscv/c1_Runtime1_riscv.cpp

```diff
@@ -925,6 +925,51 @@ OopMapSet* Runtime1::generate_code_for(C1StubId id, StubAssembler* sasm) {
       }
       break;
 
+    case C1StubId::is_instance_of_id:
+      {
+        // Mirror: c_rarg0
+        // Object: c_rarg1
+        // Temps: x13, x14, x15, x16, x17
+        // Result: x10
+
+        // Get the Klass* into c_rarg6
+        Register klass = c_rarg6, obj = c_rarg1, result = x10;
+        __ ld(klass, Address(c_rarg0, java_lang_Class::klass_offset()));
+
+        Label fail, is_secondary, success;
+
+        __ beqz(klass, fail); // Klass is null
+        __ beqz(obj, fail); // obj is null
+
+        __ lwu(x13, Address(klass, in_bytes(Klass::super_check_offset_offset())));
+        __ mv(x17, in_bytes(Klass::secondary_super_cache_offset()));
+        __ beq(x13, x17, is_secondary); // Klass is a secondary superclass
+
+        // Klass is a concrete class
+        __ load_klass(x15, obj);
+        __ add(x17, x15, x13);
+        __ ld(x17, Address(x17));
+        __ beq(klass, x17, success);
+        __ mv(result, 0);
```

Comment:
This is also a failure case?
Could it bring some benefit to jump to `fail` below or put the `fail` target here (i.e. remove the `fail` below)?

Reply:
> This is also a failure case? Could it bring some benefit to jump to `fail` below or put the `fail` target here (i.e. remove the `fail` below)?

Thanks for your review. In fact I don't know if this is a failure case here, I tried jumping to success or failure first(beq(klass, x17, success)/bne(klass, x17, fail)) and the performance results had little impact.